File: typescript/fixtures/broken-syntax/bad_generics.ts
// DO NOT FIX. Intentionally invalid: unbalanced generics, truncated extends constraint.

export class Holder<T, U {
    constructor(
        readonly left: T,
        readonly right: U,
    ) {}
}

export function merge<T extends>(left: T, right: T): Array<T {
    return [left, right];
}

File: rust/fixtures/broken-syntax/bad_generics.rs
// DO NOT FIX. Intentionally invalid: unbalanced generics, truncated where clause.

pub struct Holder<T, U {
    pub left: T,
    pub right: U,
}

pub fn merge<T>(left: T, right: T) -> Vec<T
where
    T:
{
    vec![left, right]
}
